api/all/src/main/java/io/opentelemetry/api/baggage/propagation/Element.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.baggage.propagation;
+
+import java.util.BitSet;
+
+class Element {
+
+  private final BitSet excluded = new BitSet(128);
+
+  private boolean leadingSpace;
+  private boolean readingValue;
+  private boolean trailingSpace;
+  private int start;
+  private int end;
+  private String value;
+
+  Element(char[] excludedChars) {
+    for (char separator : excludedChars) {
+      excluded.set(separator);
+    }
+    reset(0);
+  }
+
+  String getValue() {
+    return value;
+  }
+
+  void reset(int start) {
+    this.start = start;
+    leadingSpace = true;
+    readingValue = false;
+    trailingSpace = false;
+    value = null;
+  }
+
+  boolean tryTerminating(int i, String header) {
+    if (this.readingValue) {
+      markEnd(i);
+    }
+    if (this.trailingSpace) {
+      setValue(header);
+      return true;
+    } else {
+      // leading spaces - no content, invalid
+      return false;
+    }
+  }
+
+  private void markEnd(int end) {
+    this.end = end;
+    this.readingValue = false;
+    trailingSpace = true;
+  }
+
+  private void setValue(String header) {
+    this.value = header.substring(this.start, this.end);
+  }
+
+  boolean tryNextChar(char character, int i) {
```

Comment:
this method is always called with a `!` in front of it. It might be better to invert the logic and rename the method appropriately.

Reply:
Initially I did exactly as you suggest, but then the method returns "false" if it completes properly (so "try" is successful).  I found this highly misleading, so changed to what we have now. All of the usages read now: "if try is unsuccessful, then...". Makes sense?

Reply:
you could rename it to something like "rejectNextChar"?